### PR TITLE
Dataproc: support pyfiles in zip, and --region option

### PR DESCRIPTION
**Pull request by vladsaveliev**
A few adjustments to `hailctl dataproc` relevant mostly for us for the time we are using dataproc (allowing a zip archive as pyfiles, and add the `--region` option). Examples in the joint-calling workflow [README](https://github.com/populationgenomics/joint-calling-workflow) rely on these changes.

### CODE REVIEW

File: hail/python/hailtop/hailctl/dataproc/stop.py [hunk 1]
```diff
@@ -5,14 +5,22 @@ def init_parser(parser):
...
 
 def main(args, pass_through_args):
     print("Stopping cluster '{}'...".format(args.name))
 
-    cmd = ['dataproc', 'clusters', 'delete', '--quiet', args.name]
+    cmd = [
+        'dataproc', 
+        'clusters', 
+        'delete', 
+        '--region={}'.format(args.region),
```
Comment (illusional) [inline hail/python/hailtop/hailctl/dataproc/stop.py:20]: Why not an f-string here?

```suggestion
       f'--region={args.region}'
```

File: hail/python/hailtop/hailctl/dataproc/submit.py [hunk 2]
```diff
@@ -22,27 +23,32 @@ def main(args, pass_through_args):  # pylint: disable=unused-argument
...
-                for root, _, pyfiles_walk in os.walk(hail_script_entry):
-                    for pyfile in pyfiles_walk:
-                        if pyfile.endswith('.py'):
-                            zipf.write(os.path.join(root, pyfile),
-                                       os.path.relpath(os.path.join(root, pyfile),
-                                                       os.path.join(hail_script_entry, '..')))
-        zipf.close()
-        pyfiles = tfile
+
+    if args.pyfiles and args.pyfiles.endswith('.zip') and not ',' in args.pyfiles:
```
Comment (illusional) [inline hail/python/hailtop/hailctl/dataproc/submit.py:27]:
```suggestion
    # If you only provide one (comma-sep) argument, and it's a zip file, use that file directly
    if args.pyfiles and args.pyfiles.endswith('.zip') and not ',' in args.pyfiles:
```

**Review by illusional — APPROVED** [review]
Looks good! I think it'd be worth adding a small comment before the alternative pathway for zip :)

File: hail/python/hailtop/hailctl/dataproc/stop.py (hunk 1 above)
Comment (vladsaveliev) [reply hail/python/hailtop/hailctl/dataproc/stop.py:20]: Totally agree about f-strings in general, but this one was just for consistency with `submit.py`

**Comment by vladsaveliev** [comment]
Thank you Michael!

**Pull request merged**